lms/djangoapps/course_blocks/transformers/tests/test_user_partitions.py

```diff
@@ -212,7 +217,203 @@ class MergedGroupAccessTestCase(UserPartitionTestMixin, CourseStructureTestCase)
     """
     _MergedGroupAccess Test
     """
-    # TODO Test Merged Group Access (MA-1624)
+    def setUp(self):
+        """
+        Setup course structure and create user for user partition
+        transformer test.
+        """
+        super(MergedGroupAccessTestCase, self).setUp()
+
+        # Set up multiple user partitions and groups.
+        self.setup_groups_partitions(num_user_partitions=3)
+
+        self.course = CourseFactory.create(user_partitions=self.user_partitions)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id, is_active=True)
+
+        # Set up cohorts.
+        self.setup_cohorts(self.course)
+
+    def get_course_hierarchy(self):
+        """
+        Returns a course hierarchy to test with.
+        """
+        # The block tree is as follows, with the numbers in the brackets
+        # specifying the group_id for each of the 3 partitions.
+        #               A
+        #        /      |    \
+        #       /       |     \
+        #     B         C      D
+        # [1][3][]  [2][2][]  [3][1][]
+        #      \    /
+        #       \  /
+        #         E
+        #
+        return [
+            {
+                'org': 'MergedGroupAccess',
+                'course': 'MGA101F',
+                'run': 'test_run',
+                'user_partitions': self.user_partitions,
+                '#type': 'course',
+                '#ref': 'A',
+            },
+            {
+                '#type': 'vertical',
+                '#ref': 'B',
+                '#parents': ['A'],
+                'metadata': {'group_access': {1: [1], 2:[3], 3:[]}},
+            },
+            {
+                '#type': 'vertical',
+                '#ref': 'C',
+                '#parents': ['A'],
+                'metadata': {'group_access': {1: [2], 2:[2], 3:[]}},
+            },
+            {
+                '#type': 'vertical',
+                '#ref': 'D',
+                '#parents': ['A'],
+                'metadata': {'group_access': {1: [3], 2:[1], 3:[]}},
+            },
+            {
+                '#type': 'vertical',
+                '#ref': 'E',
+                '#parents': ['B', 'C'],
+            },
+        ]
+
+    AccessTestCase = namedtuple(
```

Comment:
Not a TestCase (that term has a particular meaning in python's unittest framework). Maybe rename `AccessTestData`?